src/consensus/validation.h

```diff
@@ -93,8 +93,10 @@ class CValidationState {
 // using only serialization with and without witness data. As witness_size
 // is equal to total_size - stripped_size, this formula is identical to:
 // weight = (stripped_size * 3) + total_size.
-static inline int64_t GetTransactionWeight(const CTransaction& tx)
+template <typename T>
+static inline int64_t GetTransactionWeight(const T& tx)
 {
+    static_assert(std::is_same<T, CTransaction>::value || std::is_same<T, CMutableTransaction>::value, "no need to allow other types");
```

Comment:
it might be nicer to have this function not participate in overload resolution (i.e. `enable_if`) when these conditions are not met, rather than being triggering a static assert.

Reply:
Downside of using enable_if is that the resulting compiler error message will probably be more ambiguous and harder to understand.